CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,432 @@
+                      leadingTrivia: .newline,
+                      keyExpression: ExprSyntax(#""\#(raw: child.swiftName)""#),
+                      valueExpression: ExprSyntax("\(raw: child.swiftName).map(Syntax.init)?.asProtocol(SyntaxProtocol.self) as Any"))
+                  } else {
+                    DictionaryElementSyntax(
+                      leadingTrivia: .newline,
+                      keyExpression: ExprSyntax(#""\#(raw: child.swiftName)""#),
+                      valueExpression: ExprSyntax("Syntax(\(raw: child.swiftName)).asProtocol(SyntaxProtocol.self)"))
+                  }
+                }
+              }
+            )
+          })
+        }
+      }
+    }
+  }
+}
+
+private func generateSyntaxChildChoices(for child: Child) throws -> EnumDeclSyntax {
+  return try EnumDeclSyntax("public enum \(raw: child.name): SyntaxChildChoices") {
+    for choice in child.nodeChoices {
+      DeclSyntax("case `\(raw: choice.swiftName)`(\(raw: choice.typeName))")
+    }
+
+    try! VariableDeclSyntax("public var _syntaxNode: Syntax") {
+      try! SwitchStmtSyntax("switch self") {
+        for choice in child.nodeChoices {
+          SwitchCaseSyntax("case .\(raw: choice.swiftName)(let node):") {
+            StmtSyntax("return node._syntaxNode")
+          }
+        }
+      }
+    }
+
+    DeclSyntax("init(_ data: SyntaxData) { self.init(Syntax(data))! }")
+
+    for choice in child.nodeChoices {
+      if let choiceNode = SYNTAX_NODE_MAP[choice.syntaxKind], choiceNode.isBase {
+        DeclSyntax("""
+          public init<Node: \(raw: choiceNode.name)Protocol>(_ node: Node) {
+            self = .\(raw: choice.swiftName)(\(raw: choiceNode.name)(node))
+          }
+          """)
+
+      } else {
+        DeclSyntax("""
+          public init(_ node: \(raw: choice.typeName)) {
+            self = .\(raw: choice.swiftName)(node)
+          }
+          """)
+      }
+    }
+
+    try! InitializerDeclSyntax("public init?<S: SyntaxProtocol>(_ node: S)") {
+      for choice in child.nodeChoices {
+        StmtSyntax("""
+          if let node = node.as(\(raw: choice.typeName).self) {
+            self = .\(raw: choice.swiftName)(node)
+            return
+          }
+          """)
+      }
+
+      StmtSyntax("return nil")
+    }
+
+    try! VariableDeclSyntax("public static var structure: SyntaxNodeStructure") {
+      let choices = ArrayExprSyntax {
+        for choice in child.nodeChoices {
+          ArrayElementSyntax(
+            expression: ExprSyntax(".node(\(raw: choice.typeName).self)"))
+        }
+      }
+
+      StmtSyntax("return .choices(\(choices))")
+    }
+  }
+}
+
+private func generateInitializerDecl(children: [Child], optionalBaseAsMissing: Bool, currentIndentation: String = "") -> DeclSyntax {
+    if children.isEmpty {
+        return "public init()"
+    }
+
+    var genericParamNames: [String: Int] = [:]
+    var genericParams: [String] = []
+    var params = ["leadingTrivia: Trivia? = nil"]
```

Comment:
Not part of this PR but maybe this could be a reason to make `FunctionParameterSyntax` `ExpressibleByStringInterpolation` so `params` can be of type `[FunctionParameterSyntax]`.

Reply:
Did the same for he generic parameters

Reply:
Nice.